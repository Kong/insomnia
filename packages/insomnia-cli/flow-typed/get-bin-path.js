// @flow

declare module 'get-bin-path' {
  declare module.exports: *;
}
