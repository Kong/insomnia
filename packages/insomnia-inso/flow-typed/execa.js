// @flow

declare module 'execa' {
  declare module.exports: *;
}
