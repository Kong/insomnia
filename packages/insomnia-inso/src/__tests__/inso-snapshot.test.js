// @flow
import execa from 'execa';
import { getBinPathSync } from 'get-bin-path';

// MAKE SURE YOU BUILD THE PROJECT BEFORE RUNNING THESE TESTS.
// These tests use the executable /bin/inso, which relies on /dist.

describe('Snapshot for', () => {
  it.each([
    '-h',
    '--help',
    'help',
    'generate -h',
    'generate config -h',
    'run -h',
    'run test -h',
    'lint -h',
    'lint spec -h',
    'export -h',
    'export spec -h',
  ])(
    '"inso %s"',
    async args => {
      const { stdout } = await execa(getBinPathSync(), args.split(' '));
      expect(stdout).toMatchSnapshot();
    },
    30000,
  );
});
