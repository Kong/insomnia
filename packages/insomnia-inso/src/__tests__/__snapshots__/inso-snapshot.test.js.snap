// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for "inso --help" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version             output the version number
  -w, --working-dir <dir>   set working directory
  -a, --app-data-dit <dir>  set the app data directory
  -h, --help                display help for command

Commands:
  generate                  Code generation utilities
  run                       Execution utilities
  lint                      Linting utilities
  help [command]            display help for command"
`;

exports[`Snapshot for "inso -h" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version             output the version number
  -w, --working-dir <dir>   set working directory
  -a, --app-data-dit <dir>  set the app data directory
  -h, --help                display help for command

Commands:
  generate                  Code generation utilities
  run                       Execution utilities
  lint                      Linting utilities
  help [command]            display help for command"
`;

exports[`Snapshot for "inso generate -h" 1`] = `
"Usage: inso generate [options] [command]

Code generation utilities

Options:
  -h, --help                     display help for command

Commands:
  config [options] [identifier]  Generate configuration from an api spec.
  help [command]                 display help for command"
`;

exports[`Snapshot for "inso generate config -h" 1`] = `
"Usage: inso generate config [options] [identifier]

Generate configuration from an api spec.

Options:
  -t, --type <value>   type of configuration to generate, options are
                       [kubernetes, declarative] (default: \\"declarative\\")
  -o, --output <path>  save the generated config to a file
  -h, --help           display help for command"
`;

exports[`Snapshot for "inso help" 1`] = `
"Usage: inso [options] [command]

A CLI for Insomnia!

Options:
  -v, --version             output the version number
  -w, --working-dir <dir>   set working directory
  -a, --app-data-dit <dir>  set the app data directory
  -h, --help                display help for command

Commands:
  generate                  Code generation utilities
  run                       Execution utilities
  lint                      Linting utilities
  help [command]            display help for command"
`;

exports[`Snapshot for "inso lint -h" 1`] = `
"Usage: inso lint [options] [command]

Linting utilities

Options:
  -h, --help         display help for command

Commands:
  spec [identifier]  Lint an API Specification
  help [command]     display help for command"
`;

exports[`Snapshot for "inso lint spec -h" 1`] = `
"Usage: inso lint spec [options] [identifier]

Lint an API Specification

Options:
  -h, --help  display help for command"
`;

exports[`Snapshot for "inso run -h" 1`] = `
"Usage: inso run [options] [command]

Execution utilities

Options:
  -h, --help      display help for command

Commands:
  test [options]  Run Insomnia unit tests
  help [command]  display help for command"
`;

exports[`Snapshot for "inso run test -h" 1`] = `
"Usage: inso run test [options]

Run Insomnia unit tests

Options:
  -r, --reporter <reporter>  reporter to use, options are [dot, list, spec,
                             min, progress] (default: \\"spec\\")
  -b, --bail                 abort (\\"bail\\") after first test failure
  --keep-file                do not delete the generated test file
  -h, --help                 display help for command"
`;
